### ISSUE TITLE
It is not supporting ts 

### ISSUE THREAD

**Issue by himanshi2910** [paul-shuvo/nodejs-currency-converter#67]
Could not find a declaration file for module 'currency-converter-lt'. '/node_modules/currency-converter-lt/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/currency-converter-lt` if it exists or add a new declaration (.d.ts) file containing `declare module 'currency-converter-lt';`ts(7016)

After installing `npm i --save-dev @types/currency-converter-lt`, getting this error: 
npm i --save-dev @types/currency-converter-lt
Password:
npm error code E404
npm error 404 Not Found - GET https://registry.npmjs.org/@types%2fcurrency-converter-lt - Not found
npm error 404
npm error 404  '@types/currency-converter-lt@*' is not in this registry.
npm error 404
npm error 404 Note that you can also install from a
npm error 404 tarball, folder, http url, or git url.

**Comment by The-White-Fang** [comment]
Assuming you are using `src` directory

1. Create a path `src/@types/currency-converter-lt/index.d.ts`.
2. Paste the contents from [this gist](https://gist.github.com/The-White-Fang/d8a92398510bbfa0ed195e7caa2a99e0)
3. Update `tsconfig.json` to include types from `src/@types`. 
`"typeRoots": ["src/@types", "node_modules/@types"],`

**Comment by olarotseyi** [comment]
Did you try to install it normally using `npm i currency-converter-lt`?

**Comment by The-White-Fang** [comment]
@olarotseyi I don't think that would solve the issue. Typescript requires type definitions to be present in the modules being used. And since the `currency-converter-lt` does not have any, it throws an error. This can be fixed by providing the types using `.d.ts` files. I can create a `@types/currency-converter-lt` package if someone needs it, I also have provided a gist in case you want to add it to the package itself.

**Comment by olarotseyi** [comment]
@The-White-Fang Yes, you are right. Cloning this repository and creating a separate package for typescript might be the best option so that it doesn't interrupt people already using the package in javascript projects. I'm sure people using typescript like @himanshi2910 will appreciate the effort.

**Comment by olarotseyi** [comment]
@The-White-Fang If you clone this repo to create a separate package, you can update the ReadMe in the new package to state that the new package was created from this repo so that the new repo can start off with some level of trust.

**Comment by The-White-Fang** [comment]
Hey @olarotseyi, I noticed that this repo hasn't been updated for quite some time. I am willing to work on it as the current project requires this to work, but it will be pointless if the fixes aren't merged. Do you have the merge rights? If yes, I will create some PRs within the next week. If not, publishing a new package will be more reasonable.

**Comment by olarotseyi** [comment]
If your PR won’t affect users currently using the package with JavaScript, then you can create the PRs to support Typescript and @paul-shuvo will merge it if no conflict otherwise, I think it’s best to clone the repo and make it adaptable to typescript. I wonder if @himanshi2910 was able to try the suggestion made by @The-White-Fang on Aug 24, 2024 regarding this issue. It would be nice to know if @himanshi2910 tried the suggestion and if it solved the problem.
